agent/app/agent.go

```diff
@@ -402,6 +404,10 @@ func (agent *ecsAgent) registerContainerInstance(
 			seelog.Critical("Instance registration attempt with an invalid attribute")
 			return err
 		}
+		awsErr, ok := err.(awserr.Error)
```

Comment:
This should be before `utils.AttributeError` comparison. Also, a minor nit here: `if awsErr, ok := ...` pattern leads to lesser locs.

Reply:
done.